api/v1/model/src/commonMain/kotlin/JobConfigurations.kt

```diff
@@ -73,6 +73,12 @@ data class AnalyzerJobConfiguration(
      */
     val environmentConfig: EnvironmentConfig? = null,
 
+    /**
+     * A flag indicating whether the Git submodules of the repository should be ignored during the download process.
+     * If set to `true`, submodules will not be downloaded; if `false`, they will be included.
+     */
+    val ignoreSubmodules: Boolean = false,
```

Comment:
Practically we only see people using Git nowadays, but theoretically this could also be a Mercurial, SVN, or Git-Repo repository, so I would prefer a name that is not so Git specific. Maybe `recursiveCheckout` to align with the `recursive` paramter of the `updateWorkingTree` function.

Reply:
Onur already did a refactoring to the term _recursiveCheckout_